hubstorage/resourcetype.py

```diff
@@ -26,7 +28,9 @@ def _iter_lines(self, _path, **kwargs):
 
         r = self.client.request(**kwargs)
 
-        return r.iter_lines()
+        lines = r.iter_lines()
+        return (l.decode('utf8') if isinstance(l, six.binary_type)
+                else l for l in lines)
```

Comment:
> I deleted the branch and pulled it again - the issue persists

I apologize for that. I was using an experimental Hubstorage branch that was returning back arbitrary results. I was able to reproduce the issue.

I agree we should keep backwards compatibility in py2. I've pushed a changed [1] to handle that.

I've also added a unit test [2] to make sure the output is compatible with the respective python version.

> how so? Content-Type: application/x-jsonlines; charset=UTF-8

If I set `decode_unicode=True` in py2 I get a generator that produces unicode strings, as expected. However, in py3 the generator sometimes produces bytes strings and sometimes unicode strings.

Inspecting `r.encoding` here [3] I noticed that it's occasionally set as `None`. I don't know if this is an issue with `requests` or Hubstorage (or if it's an issue at all), but I can't rely only on `decode_unicode=True` because it doesn't guarantee [4] the decoding.

[1] https://github.com/scrapinghub/python-hubstorage/blob/cece2665987ebc7acdf1471ea3a4cdc7cc9e7428/hubstorage/resourcetype.py#L32-L34
[2] https://github.com/scrapinghub/python-hubstorage/pull/62/commits/cece2665987ebc7acdf1471ea3a4cdc7cc9e7428
[3] https://github.com/scrapinghub/python-hubstorage/blob/31525d31c536eacfc19f416404e64ab21598f1d2/hubstorage/resourcetype.py#L27
[4] http://docs.python-requests.org/en/master/api/#requests.Response.iter_content